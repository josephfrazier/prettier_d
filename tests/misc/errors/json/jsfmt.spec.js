run_spec(
  {
    dirname: __dirname,
    snippets: [
      "{foo}",
      '{["foo"]:"bar"}',
      '{"foo": ~1}',
      '{"foo": false || "bar"}',
      '{"foo": undefined}',
      '{"foo": () => {}}',
    ],
  },
  ["json-stringify"]
);
