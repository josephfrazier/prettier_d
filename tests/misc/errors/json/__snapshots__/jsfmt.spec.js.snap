// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snippet: #0 error test 1`] = `
"ObjectProperty with shorthand=true is not allowed in JSON. (1:2)
> 1 | {foo}
    |  ^"
`;

exports[`snippet: #1 error test 1`] = `
"ObjectProperty with computed=true is not allowed in JSON. (1:2)
> 1 | {[\\"foo\\"]:\\"bar\\"}
    |  ^"
`;

exports[`snippet: #2 error test 1`] = `
"UnaryExpression with operator=\\"~\\" is not allowed in JSON. (1:9)
> 1 | {\\"foo\\": ~1}
    |         ^"
`;

exports[`snippet: #3 error test 1`] = `
"LogicalExpression is not allowed in JSON. (1:9)
> 1 | {\\"foo\\": false || \\"bar\\"}
    |         ^"
`;

exports[`snippet: #4 error test 1`] = `
"Identifier is not allowed in JSON. (1:9)
> 1 | {\\"foo\\": undefined}
    |         ^"
`;

exports[`snippet: #5 error test 1`] = `
"ArrowFunctionExpression is not allowed in JSON. (1:9)
> 1 | {\\"foo\\": () => {}}
    |         ^"
`;
